clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -3582,21 +3407,55 @@ static bool InvalidWorkGroupSizeAttrs(const Expr *MGValue, const Expr *XDim,
            ZDimExpr->getResultAsAPSInt() != 1));
 }
 
-// If the [[intel::max_work_group_size(X, Y, Z)]] attribute is specified on
-// a declaration along with [[sycl::reqd_work_group_size(X1, Y1, Z1)]]
-// attribute, check to see if values of reqd_work_group_size arguments are
+// If the 'reqd_work_group_size' attribute is specified on
+// a declaration along with 'max_work_group_size' attribute,
+// check to see if values of reqd_work_group_size arguments are
 // equal or less than values of max_work_group_size attribute arguments.
-static bool checkWorkGroupSizeAttrValues(const Expr *RWGS, const Expr *MWGS) {
+// If the 'reqd_work_group_size' attribute is specified on
+// a declaration along with 'reqd_work_group_size' attribute,
+// check to see if values of reqd_work_group_size arguments are
+// equal or greater than values of 'reqd_work_group_size' attribute arguments.
+
```

Comment:
```
// Compare the first set of X, Y, and Z dimension values with the second set of values passed in.
// The comparison is made using the operator passed as the template parameter.
```

How about something general like the above, as opposed to describing what the parameters are supposed to be here (in terms of the attributes)?   This function right now is sufficiently general.